hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -430,16 +430,11 @@ private static <R> Option<HoodieRecord<R>> mergeIncomingWithExistingRecordWithEx
       //the record was deleted
       return Option.empty();
     }
-    if (mergeResult.getRecord() == null) {
-      // SENTINEL case: the record did not match and merge case and should not be modified
+    if (mergeResult.getRecord() == null || mergeResult == existingBufferedRecord) {
```

Comment:
if this is only for merge into and merge into only been utilized with EVENT_TIME merge mode, equals with buffered record is valid, otherwise, better use record equals here.

Reply:
Yes, the expression payload is only used for Merge Into